wpilibc/src/main/native/cpp/Tracer.cpp

```diff
@@ -27,7 +44,20 @@ void Tracer::ClearEpochs() {
 
 void Tracer::AddEpoch(std::string_view epochName) {
   auto currentTime = hal::fpga_clock::now();
-  m_epochs[epochName] = currentTime - m_startTime;
+  auto epoch = currentTime - m_startTime;
+  m_epochs[epochName] = epoch;
+  if (m_publishNT) {
+    auto topic = nt::NetworkTableInstance::GetDefault().GetIntegerTopic(
+        m_ntTopic.data() + "/" + epochName.data());
+    auto pub = topic.Publish();
+    pub.SetDefault(0);
+    topic.SetRetained(true);
+    pub.Set(epoch.count());
+  }
+  if (m_dataLogEnabled) {
+    m_dataLog.AppendInteger(
+        m_dataLog.Start(m_dataLogEntry + "/" + epochName, "int64"), epoch, 0);
```

Comment:
`epochName` is a `string_view`, which can't be added to a string

Reply:
But you should only be starting the data log once; this seems to start it on every call?

Reply:
I’m “starting” an entry every call; I’m using it to get the entry index for an epoch so I can append the epoch to its corresponding entry. The alternative was storing the epoch names and entry indices in variables in the Tracer object, which I didn’t like. If there’s a better way, let me know.

Reply:
There's two issues:
- You're calling Start() but never Finish(); there's an internal ref count that will eventually overflow if you're only calling Start().
- You're paying the penalty for the string concatenation/allocation every call.  If you cache the name to entry ID you can avoid this (use `wpi::StringMap`).